eth/vm/base.py

```diff
@@ -438,7 +452,7 @@ def build_state(cls,
         even if you don't have the VM initialized. This is a convenience method to do that.
         """
 
-        execution_context = header.create_execution_context(previous_hashes)
+        execution_context = header.create_execution_context(previous_hashes, chain_context)
```

Comment:
Hm, by the name `ChainContext`, it seems like `previous_hashes` could be a part of it. What does it look like if you try to combine the two?

Reply:
Other than that question, approach seems fine. :+1:

Reply:
Unclear to me what previous hashes is supposed to be for other than a pass-through to `ExecutionContext` but I do agree that `ChainContext` should inform `ExecutionContext` with any valid data from outside the VM.

Is `previous_hashes` used to obtain the result of calling the `BLOCKHASH` opcode?

Reply:
:+1: to moving `previous_hashes` to `ChainContext`

Reply:
So, this doesn't seem like a good idea because the list of previous hashes needs to be generated on the spot when an execution context is started because the block height may have changed.

Perhaps `ChainContext` isn't the right way to keep track of this, maybe something more like `BlockContext` or something would be better? Also, would be ideal for `chain_id` because there is some small likelihood that this number would change at a particular block height for a contentious fork, or if a process is invented to cycle chain ID on any hard fork.

Reply:
Fair enough, making `ChainContext` something that is immutable through the lifetime of a chain is a reasonable choice. Let's write that explicitly in the class docstring.

Reply:
Well, that's my point... chain ID _might not_ be the same for the entirety of the chain, but I think it's something that is liable to be immutable per VM. But it's _defined_ from the chain config... so confusing, lol.